File: src/3158.find-the-xor-of-numbers-which-appear-twice.3428/solution.ts
function duplicateNumbersXOR (nums: number[]): number {
    const counts = new Array<boolean>(50);
    let result = 0;
    for (const n of nums) {
        if (counts[n - 1]) {
            result ^= n;
        }
        counts[n - 1] = true;
    }
    return result;
}

File: src/3158.find-the-xor-of-numbers-which-appear-twice.3428/solution.java
class Solution {
    public int duplicateNumbersXOR(int[] nums) {
        int[] counts = new int[51];
        for(int n:nums){
            counts[n]++;
        }
        int result = 0;
        for(int i=1;i<counts.length;i++){
            if(counts[i] == 2){
                result ^= i;
            }
        }
        return result;
    }
}